core/message_thread_manager.py

```diff
@@ -0,0 +1,176 @@
+import sqlite3
+import json
+from typing import List, Dict, Any, Optional
+from core.utils.llm import make_llm_api_call
+from core.tools.files_tool import FilesTool
+
+
+class MessageThreadManager:
+    def __init__(self, db_path: str = "db.db"):
+        self.db_path = db_path
+        self.conn = sqlite3.connect(self.db_path)
+        self.cursor = self.conn.cursor()
```

Comment:
In WorkingMemory cursor is created on every interaction, here it's stored. I feel like probably one of those will be more correct and should be used in all. Also it will probably be better to pass the conn into the constructor for [inversion of control](https://martinfowler.com/bliki/InversionOfControl.html) / testability.